src/master/master.ts

```diff
@@ -303,13 +299,18 @@ export class Master {
       this.storageAPI.setState(key, stateWithoutDeltalog, deltalog);
       if (newMetadata) this.storageAPI.setMetadata(key, newMetadata);
     } else {
-      const writes = [
-        this.storageAPI.setState(key, stateWithoutDeltalog, deltalog),
-      ];
-      if (newMetadata) {
-        writes.push(this.storageAPI.setMetadata(key, newMetadata));
+      try {
+        const writes = [
+          this.storageAPI.setState(key, stateWithoutDeltalog, deltalog),
+        ];
+        if (newMetadata) {
+          writes.push(this.storageAPI.setMetadata(key, newMetadata));
+        }
+
+        await Promise.all(writes);
+      } catch (error) {
+        console.error('Error setting state:', error);
```

Comment:
issue: we are stoping the propagation of the exception here, might make the consumer understand that the process finished correctly, when that might not be the case.

Reply:
that's right, i fixed it